src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -187,5 +190,102 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
                 FileSystemAclExtensions.SetAccessControl(fileStream, fileSecurity);
             }
         }
+
+        [Fact]
+        public void DirectoryInfo_Create_NullDirectoryInfo()
+        {
+            DirectoryInfo info = null;
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<ArgumentNullException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NullDirectorySecurity()
+        {
+            DirectoryInfo info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>(() => info.Create(null));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NotFound()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<DirectoryNotFoundException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_AccessDenied()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"C:\\Windows\\System32\\deniedaccess");
```

Comment:
Will this pass if I run the test as admin? Maybe it should bail out if `PlatformDetection.IsWindowsAndElevated` is true.

Reply:
I'm not sure. Should I expect these unit tests to always be executed in elevated mode at some point, or should I indicate that condition among the test method attributes? This method in particular should not be run in elevated mode, but I could add one that does, and I would have to investigate the expected result.

Reply:
Whatever our official runs do (I assume none are), all our tests should pass whether I run elevated or not. There are some tests that require elevation and are attributed `[ConditionalFact(nameof(IsProcessElevated))]`. This may be an unusual reverse case. Not suggesting you have to verify it but an `if` would do no harm.